server/src/test/java/org/elasticsearch/client/transport/TransportClientTests.java

```diff
@@ -64,13 +67,25 @@ public void testPluginNamedWriteablesRegistered() {
         }
     }
 
-    public void testDefaultHeaderContainsPlugins() {
-        Settings baseSettings = Settings.builder()
+    public void testSettingsContainsTransportClient() {
+        final Settings baseSettings = Settings.builder()
             .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir())
             .build();
         try (TransportClient client = new MockTransportClient(baseSettings, Arrays.asList(MockPlugin.class))) {
-            ThreadContext threadContext = client.threadPool().getThreadContext();
-            assertEquals("true", threadContext.getHeader("transport_client"));
+            final Settings settings = TcpTransport.DEFAULT_FEATURES_SETTING.get(client.settings());
+            assertThat(settings.keySet(), hasItem("transport_client"));
+            assertThat(settings.get("transport_client"), equalTo("true"));
+            final ThreadContext threadContext = client.threadPool().getThreadContext();
+            assertEquals("true", threadContext.getHeader("test"));
```

Comment:
what is this line testing? not relevant to this test method?

Reply:
Sorry, I meant to remove that line and the line above it from this test as that functionality is covered in `testDefaultHeader` (see below). I pushed 82ed4d4ecb7a3bf842f1acddec6a267592e1c57e.